src/main/java/nl/tudelft/ewi/devhub/server/database/entities/notifications/Notification.java

```diff
@@ -0,0 +1,51 @@
+package nl.tudelft.ewi.devhub.server.database.entities.notifications;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import nl.tudelft.ewi.devhub.server.database.entities.User;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.util.Date;
+
+/**
+ * Created by Arjan on 7-6-2017.
+ */
+@Data
+@Entity
+@Table(name = "notification")
+@EqualsAndHashCode(of = {"id"})
+public class Notification {
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(name = "event")
+    @NotNull
+    //Change this to Enum?
+    private String event;
+
+    @Column(name = "message")
+    @NotEmpty
+    private String message;
+
+    @Column(name = "link")
+    // Change this to URI later
+    private String link;
```

Comment:
There already is such an interface, but its poorly named, `Base`. It requires implementations to implement a `getURI` method that returns the `URI` for its page within DevHub. Its implemented for example by 
[`AbstractIssue`](https://github.com/devhub-tud/devhub/blob/master/src/main/java/nl/tudelft/ewi/devhub/server/database/entities/issues/AbstractIssue.java#L58), [`RepositoryEntity`](https://github.com/devhub-tud/devhub/blob/master/src/main/java/nl/tudelft/ewi/devhub/server/database/entities/RepositoryEntity.java#L58) and [`Assignment`](https://github.com/devhub-tud/devhub/blob/master/src/main/java/nl/tudelft/ewi/devhub/server/database/entities/Assignment.java#L60) so that users of these entities can easily find their respective page within DevHub. I remember helping @cashenkes to implement it in [`Comment`](https://github.com/devhub-tud/devhub/blob/ef4a592a47cd13c478b0ab1b030252d4ec8eeac6/src/main/java/nl/tudelft/ewi/devhub/server/database/entities/comments/Comment.java#L36) too, so that clicking the comment notification would bring you to the right issue, pull request or commit (depending on whether it was a [`CommitComment`](https://github.com/devhub-tud/devhub/blob/master/src/main/java/nl/tudelft/ewi/devhub/server/database/entities/comments/CommitComment.java) or an [`IssueComment`](https://github.com/devhub-tud/devhub/blob/master/src/main/java/nl/tudelft/ewi/devhub/server/database/entities/comments/IssueComment.java). I can't seem to find that piece of code back in the repository though.

Anyways, implementing this interface for your `Notification` class is not trivial. Because what would be the implementation of this `getURI` method? Implementing the interface does not solve the issue addressed above: that storing the URI at persist time is not a great idea, and the notification should instead be linked to the entity it references.

However, the entities it may reference are from different types (so far `Issues` and `PullRequests`, but presumably `Commits`, `Deliveries`, and `Comments` in the near future too). Not only are these different types, but they are stored in different tables as well. As there is no common ancestor (besides `java.lang.Object`) there is no way to map all these types to a single field. So if you want to have the option to reference any of these types, you find your self doing something like this:

```java
private Issue linkedIssue;
private PullRequest linkedPullRequest;
// Many others to come...

public URI getURI() {
    if (linkedIssue != null) return linkedIssue.getURI();
    if (linkedPullRequest != null) return linkedPullRequest.getURI();
    // Many others to come...
}
``` 

This is where you end up having a lot of fields which lead to an ambiguous API. What should happen if I for example do:

```java
Notification notification = new Notification();
notification.setPullRequest(pullRequestB);
notification.setLinkedIssue(issueA);
```

A user may expect that setting the linked issue overrides the previously set pull request. But as observed from the code for `getURI` above, we can see that if there is a linkedIssue, its URI will be returned, ignoring any value in the other fields.

So we don't want this. It would be preferable instead to have subtypes of `Notifications` (for example `PullRequestNotification` that references to a `PullRequest`.

By the way, keep in mind that if we take this route, there is no need to distinguish between `PullRequestNotifications` and `IssueNotifications` because in this special case, `PullRequest` and `Issue` share a queryable common ancestor: `AbstractIssue`.